examples/WordServer.hs

```diff
@@ -20,17 +22,21 @@ fetch w = threadDelay 1000000 >> return (w,w)
 -- connection is closed.
 lookupWords :: Socket -> IO ()
 lookupWords sk =
-      Stream.unfold Socket.read sk               -- SerialT IO Word8
-    & Unicode.decodeLatin1                       -- SerialT IO Char
-    & Stream.wordsBy isSpace Fold.toList         -- SerialT IO String
-    & Stream.fromSerial                          -- AheadT  IO String
-    & Stream.mapM fetch                          -- AheadT  IO (String, String)
-    & Stream.fromAhead                           -- SerialT IO (String, String)
-    & Stream.map show                            -- SerialT IO String
-    & Stream.intersperse "\n"                    -- SerialT IO String
-    & Unicode.encodeStrings Unicode.encodeLatin1 -- SerialT IO (Array Word8)
+      Stream.unfold Socket.reader sk             -- Stream IO Word8
+    & Unicode.decodeLatin1                       -- Stream IO Char
+    & Stream.parseMany word                      -- Stream IO String
+    & Concur.mapM fetch
+    & Concur.applyWith
+        (Concur.ordered True)
+        (Stream.fromPure id)                     -- Stream IO (String, String)
```

Comment:
This is wrong, have you checked the output? mapM has already given us unordered output, how can we order it now? We should use `ordered True` in mapM.